### PR TITLE
[PORT] Removes some dead/unused code

### PR DESCRIPTION
**Pull request by imsynthex**
## About The Pull Request

Ports: https://github.com/tgstation/tgstation/pull/48158

It's in "things to port"

## Why It's Good For The Game

Has no effect on gameplay, hopefully

## Testing Photographs and Procedure

It compiles

## Changelog
:cl: spookydonut, imsynthex
code: removed some unused code
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 1]
```diff
@@ -192,7 +192,7 @@
 	genes = list(/datum/plant_gene/trait/plant_type/fungal_metabolism)
 	growing_icon = 'icons/obj/hydroponics/growing_mushrooms.dmi'
 	reagents_add = list(/datum/reagent/consumable/nutriment = 0.1)
-	mutatelist = list(/obj/item/seeds/chanterelle/jupitercup)
+	mutatelist = list(/obj/item/seeds/chanter/jupitercup)
```
Comment (itsmeow) [inline code/modules/hydroponics/grown/mushrooms.dm:195]: why repath this?

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/grown/mushrooms.dm (hunk 1 above)
Comment (imsynthex) [reply code/modules/hydroponics/grown/mushrooms.dm:195]: To make jupiter cups a subtype of chanterelles, much like every other plant and it's mutation(s)

Comment (itsmeow) [reply code/modules/hydroponics/grown/mushrooms.dm:195]: no, why did you change chantrelle to chanter. There is no reason to do that

Comment (itsmeow) [reply code/modules/hydroponics/grown/mushrooms.dm:195]: Oh I see the existing path is chanter. that is dumb

**Review by itsmeow — DISMISSED** [review]

**Comment by itsmeow** [comment]
Fails test

```
FAIL: /datum/unit_test/plantgrowth 0s
	REASON #1: seed (/obj/item/seeds/new_seed) lacks the -grow1 icon!
	REASON #2: seed (/obj/item/seeds/new_seed) lacks the -grow2 icon!
	REASON #3: seed (/obj/item/seeds/new_seed) lacks the -grow3 icon!
	REASON #4: seed (/obj/item/seeds/new_seed) lacks the -grow4 icon!
	REASON #5: seed (/obj/item/seeds/new_seed) lacks the -grow5 icon!
	REASON #6: seed (/obj/item/seeds/new_seed) lacks the -grow6 icon!
	REASON #7: seed (/obj/item/seeds/new_seed) lacks the -dead icon!
	REASON #8: seed (/obj/item/seeds/new_seed) lacks the -harvest icon!
```

**Comment by imsynthex** [comment]
Well would you look at that

**Review by itsmeow — APPROVED** [review]
Removing unit test is cringe but ok

**Pull request merged**